app/main.py

```diff
@@ -2,0 +1,28 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return str(self.__dict__).title().replace("'", "")
+
+    @classmethod
+    def get_alive(cls) -> list:
+        return [(repr(animal) for animal in cls.alive)]
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+        if herbivore.health <= 0:
+            Animal.alive.remove(herbivore)
```

Comment:
In the `bite` method, before removing the `herbivore` from `Animal.alive`, you should check if it is in the list to avoid potential errors: `if herbivore in Animal.alive:`.

Reply:
if herbivore.health <= 0 and herbivore in Animal.alive: